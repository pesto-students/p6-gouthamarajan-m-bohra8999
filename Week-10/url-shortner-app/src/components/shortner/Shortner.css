.shortner-container {
  display: flex;
  justify-content: space-between;
  padding: 50px;
  flex-direction: column;
  align-items: center;
}

@media screen and (max-width: 600px) {
  .shortner-container {
    padding: 20px;
  }
}
