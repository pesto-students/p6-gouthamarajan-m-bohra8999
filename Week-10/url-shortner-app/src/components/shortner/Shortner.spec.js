import renderer from 'react-test-renderer';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Shortner from '.';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';

const handlers = [
  rest.get('https://api.shrtco.de/v2/shorten', (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        result: {
          original_link: 'original_link',
          full_short_link: 'full_short_link',
          code: '123abcd',
        },
      })
    );
  }),
];

const server = setupServer(...handlers);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

it('renders without crashing', () => {
  render(<Shortner />);
  expect(screen.getByText('SHORTEN IT!')).toBeInTheDocument();

  const tree = renderer.create(<Shortner />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should shorten the url and display', async () => {
  render(<Shortner />);
  const button = screen.getByRole('button');
  const input = screen.getByTestId('name-input');
  fireEvent.change(input, { target: { value: 'samplelink.com' } });

  fireEvent.click(button);

  await waitFor(() => expect(screen.getAllByText('original_link')).toHaveLength(1), { timeout: 1000 });
  await waitFor(() => expect(screen.getAllByText('full_short_link')).toHaveLength(1), { timeout: 1000 });
});

test('should give proper error is request fails', async () => {
  server.use(
    rest.get('https://api.shrtco.de/v2/shorten', (req, res, ctx) => {
      return res.once(ctx.status(400), ctx.json({ message: 'Invalid URL' }));
    })
  );
  render(<Shortner />);
  fireEvent.change(screen.getByTestId('name-input'), { target: { value: 'samplelink.com' } });
  fireEvent.click(screen.getByRole('button'));
  let error = await screen.findByRole('alert');
  expect(error).toBeInTheDocument();
});
