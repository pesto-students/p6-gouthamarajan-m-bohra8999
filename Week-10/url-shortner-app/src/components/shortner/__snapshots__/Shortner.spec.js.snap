// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="shortner-container"
>
  <div
    className="input-box"
  >
    <form
      className="input-box__form"
      name="search-bar"
      onSubmit={[Function]}
    >
      <input
        className="input-box__field"
        data-testid="name-input"
        onChange={[Function]}
        placeholder="Enter link"
        type="text"
        value=""
      />
      <button
        className="input-box__action-btn"
        type="submit"
      >
        SHORTEN IT!
      </button>
    </form>
  </div>
</section>
`;
